crates/cdk/src/wallet/mod.rs

```diff
@@ -26,123 +25,59 @@ use crate::nuts::{
 use crate::types::{MeltQuote, Melted, MintQuote, ProofInfo};
 use crate::url::UncheckedUrl;
 use crate::util::{hex, unix_time};
-use crate::{Amount, Bolt11Invoice, HttpClient};
+use crate::{Amount, Bolt11Invoice, HttpClient, SECP256K1};
 
 pub mod client;
 pub mod error;
-#[cfg(feature = "nostr")]
-pub mod nostr;
 pub mod util;
 
 #[derive(Debug, Clone)]
 pub struct Wallet {
+    pub mint_url: UncheckedUrl,
+    pub unit: CurrencyUnit,
     pub client: HttpClient,
     pub localstore: Arc<dyn WalletDatabase<Err = cdk_database::Error> + Send + Sync>,
     xpriv: ExtendedPrivKey,
-    p2pk_signing_keys: Arc<RwLock<HashMap<XOnlyPublicKey, SecretKey>>>,
-    #[cfg(feature = "nostr")]
-    nostr_client: nostr_sdk::Client,
 }
 
 impl Wallet {
     pub fn new(
+        mint_url: &str,
+        unit: CurrencyUnit,
         localstore: Arc<dyn WalletDatabase<Err = cdk_database::Error> + Send + Sync>,
         seed: &[u8],
```

Comment:
Is it safe to re-use the same seed for different mint URLs?

Reply:
Yes, wallets shouldn't really care about the mint_url other then how to access the mint. When generating secrets the keyset_id is used so as long as the mints have different keysets different secrets will be generated. Similarly if a mint is exposed on two endpoints the wallet should treat it has the same mint in terms of deriving secrets.